pythran/pythonic/numpy/eye.hpp

```
@@ -2,14 +2,15 @@
 #define PYTHONIC_NUMPY_EYE_HPP
 
 #include "pythonic/numpy/zeros.hpp"
+#include "pythonic/__builtin__/None.hpp"
 
 namespace pythonic {
 
     namespace numpy {
+
         template<class dtype = double>
-            types::ndarray<typename types::numpy_type<dtype>::type, 2> eye(long N, long M=-1, long k=0, dtype d=dtype())
+            types::ndarray<typename types::numpy_type<dtype>::type, 2> eye(long N, long M, long k=0, dtype d=dtype())
             {
-                if(M<0) M = N;
```

I checked and we should throw an exception in this case.